### ISSUE TITLE
[BC 24.0] Publish-NavApp hangs forever on compilation errors

### ISSUE THREAD

**Issue by lippertmarkus** [microsoft/AL#7750]
Please include the following with each issue:

**1. Describe the bug**
Starting with BC24 `Publish-NAVApp` no longer shows an error when extension compilation fails but instead hangs forever.

**2. To Reproduce**

1. Setup a BC24 container (I used `sandbox-24.0.16410.18981-de`)
2. Publish an app that has a dependency that isn't published yet:
3. Observe that `Publish-NAVApp` hangs forever
    ```powershell
    PS > Publish-NAVApp -ServerInstance BC -Path C:\azurefileshare\ccr.app -SkipVerification
    # ... hangs forever ...
    ```
4. Compare that with a BC23 container (I used `sandbox-23.5.16502.19504-w1`) which shows the error immediately:
    ``` powershell
    PS > Publish-NAVApp -ServerInstance BC -Path C:\azurefileshare\ccr.app -SkipVerification
    
    # Publish-NAVApp : Extension compilation failed
    # error AL1024: A package with publisher 'Cosmo Consult', name 'COSMO Licensing', and a version compatible with '2.1.0.0' 
    # could not be loaded. Symbols for the requested app COSMO Licensing by Cosmo Consult   
    # 2.1.0.0 could not be found in the database.
    ```

**Note:** Because the developers need to copy and paste the code snippet, including a code snippet as a media file (i.e. .gif) is not sufficient.

**3. Expected behavior**
`Publish-NAVApp` should return an error like it did for BC 23.5

**4. Actual behavior**
`Publish-NAVApp` hangs forever

**5. Versions:**

- AL Language: N/A
- Visual Studio Code: N/A
- Business Central: 24.0.16410.18981
- List of Visual Studio Code extensions that you have installed: N/A

### Final Checklist

Please remember to do the following:

* [x] Search the issue repository to ensure you are reporting a new issue

* [x] Reproduce the issue after disabling all extensions except the AL Language extension

* [x] Simplify your code around the issue to better isolate the problem

**Comment by lippertmarkus** [comment]
@freddydk FYI

**Comment by lippertmarkus** [comment]
Also happens for BC 24.1.18927.19498

**Comment by SBalslev** [comment]
@lippertmarkus do you have event log entries and other details from the container that might help us?

**Comment by SBalslev** [comment]
if you can please also test platform 24.0.19840.0 or never. We found a timeout issue and the fix is being rolled out as we speak.

**Comment by joandrsn** [comment]
Just to say, we also experienced this on 24.2 OnPrem.
Our pipeline hanged for 22 hours because it just waits forever (Only stopped since we cancelled the build):
![image](https://github.com/microsoft/AL/assets/5766695/983c07a2-7e3c-4043-8961-c609512b51d8)

**Comment by NKarolak** [comment]
@SBalslev Is it possible you'd overlooked Markus' reply? Or do you really need further input, and if so: which?

**Comment by lippertmarkus** [comment]
@SBalslev it still happens in the newest CU

**Comment by SBalslev** [comment]
Lost track of this one, did anyone try the latest release?

**Comment by PooyaKharamesh** [comment]
@lippertmarkus have you tried it on any of the latest builds? also do you experience the same issue if the server is installed outside of a container?

would you also attach the events?

I will try to repro and get back to you.

**Comment by lippertmarkus** [comment]
Hello, I just tried on sandbox-24.4.22295.23050-us and it still happens. Although it seems that for some extensions it doesn't hang (maybe some install codeunits?). 

Here's the container and event logs: [logs.zip](https://github.com/user-attachments/files/16673379/logs.zip)

If you send me a private message I can send you the app file if needed.

**Comment by freddydk** [comment]
@lippertmarkus - are you running Powershell 7 or Powershell 5 inside the container?

**Comment by lippertmarkus** [comment]
@freddydk you're right, it works with PS7 but not with PS5. Is that something you can fix?

**Comment by freddydk** [comment]
It isn't something I can fix, but I think the server team will look at the bug.

Just be aware, that you very soon have to shift to PS7 anyway: https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/upgrade/deprecated-features-platform#support-for-powershell-50-removal 

When we start shipping v26, the bridge will likely be removed quickly.

**Comment by lippertmarkus** [comment]
@freddydk Did you get feedback from the server team?

**Comment by PatrickSchiefer** [comment]
Problem still exists in BC25
@freddydk @SBalslev any updates on this topic?

**Comment by yurim108** [comment]
Faced the same problem today with BC 25.3 OnPrem. 
The release pipeline hangs. 
An attemp to publish the app manually using PowerShell (version 5 is installed) on customer's server hangs.
Is installing PowerShell 7 a solution?
